File: ds/js/queue.js
function Queue() {
  const collection = [];

  this.push = function(element) {
    collection.push(element);
  }

  this.front = function() {
    return collection.shift();
  }

  this.size = function() {
    return collection.length;
  }

  this.isEmpty = function() {
    return collection.length === 0;
  }
}

const queue = new Queue();
queue.push(10);
queue.push(1010);
console.log(queue.front());
console.log(queue.size());
console.log(queue.front());
console.log(queue.size());

File: leetcode/learn/recursion-ii/22/queue.go
package main

type Queue struct {
	elements []string
}

func (q *Queue) Enqueue(element string) {
	q.elements = append(q.elements, element)
}

func (q *Queue) Dequeue() string {
	if len(q.elements) == 0 {
		return ""
	}

	element := q.elements[0]
	q.elements = q.elements[1:]
	return element
}

func (q *Queue) IsEmpty() bool {
	return len(q.elements) == 0
}

func NewQueue() *Queue {
	return &Queue{
		elements: make([]string, 0),
	}
}
